src/frontend/screens/WebView/index.tsx

```diff
@@ -124,6 +125,32 @@ function WebView() {
     window.api.trackScreen('WebView', { url: startUrl, runner })
   }, [startUrl, runner])
 
+  useEffect(() => {
+    if (!urlIsHpUrl(startUrl)) {
+      return
+    }
+
+    window.api.handleGoToGamePage(async (_, gameId) => {
+      const gameInfo = await getGameInfo(gameId, 'hyperplay')
+      navigate(`/gamepage/hyperplay/${gameId}`, {
+        state: { gameInfo, fromDM: false }
+      })
+    })
+
+    const removeHandleGoToGamePage = window.api.handleGoToGamePage(
+      async (_, gameId) => {
+        const gameInfo = await getGameInfo(gameId, 'hyperplay')
+        navigate(`/gamepage/hyperplay/${gameId}`, {
+          state: { gameInfo, fromDM: false }
+        })
+      }
+    )
```

Comment:
this is duplicated now

Reply:
yeah, i am really wondering if we need to remove the call here if we are doing that on the api:
```export const handleGoToGamePage = (
  onChange: (e: Electron.IpcRendererEvent, gameId: string) => void
) => {
  ipcRenderer.on('goToGamePage', onChange)
  return () => {
    ipcRenderer.removeListener('goToGamePage', onChange)
  }
}```

Reply:
we are not unsubscribing in the api fxn definition. we are retruning a callback that we can call later on to unsubscribe the listener

Reply:
So just retuning the same method should unsubscribe, no? we need a different method for that?

Reply:
Got it, pushed a fix.